### PR TITLE
fix: Missing hostAliases implementation

### LINKED ISSUE #81: Support hostAliases customization in otel collector
We can add hostAliases customization to otel collector helm chart. 

https://kubernetes.io/docs/tasks/network/customize-hosts-file-for-pods/ 

I am happy to submit a PR for this :)

### PR DESCRIPTION
**Pull request by LaurentMarchelli**
hostAliases customization added to otel collector helm chart, as requested by @ethernoy.
@dmitryax, would you please let me know if I have missed a step in the submission process.

Submitted as a patch, as the default configuration does not change any previously produced yaml.

Best regards,
Close #81

### CODE REVIEW

**Review by JaredTan95 — APPROVED** [review]

File: charts/opentelemetry-collector/templates/_pod.tpl [hunk 1]
```diff
@@ -6,6 +6,12 @@ imagePullSecrets:
 serviceAccountName: {{ include "opentelemetry-collector.serviceAccountName" . }}
 securityContext:
   {{- toYaml .Values.podSecurityContext | nindent 2 }}
+{{- if .Values.hostAliases }}
```
Comment (TylerHelmuth) [inline charts/opentelemetry-collector/templates/_pod.tpl:9]: `with` treats `[]` as false, so we can exclude the `if` statement

Comment (LaurentMarchelli) [reply charts/opentelemetry-collector/templates/_pod.tpl]: Hi @TylerHelmuth,
I removed it.
Thx

File: charts/opentelemetry-collector/values.schema.json [hunk 2]
```diff
@@ -551,6 +551,24 @@
...
+        "properties": {
+          "hostnames": {
+            "type": "array",
+            "items": {
+              "type": "string"
+            }
+          },
+          "ip": {
+            "type": "string"
+          }
```
Comment (TylerHelmuth) [inline charts/opentelemetry-collector/values.schema.json:568]:
```suggestion
```

Normally when we are planning to use the object directly in the chart we don't do our property enforcement.  This lets the feature scale and kubernetes changes.

Comment (LaurentMarchelli) [reply charts/opentelemetry-collector/values.schema.json]: Hi @TylerHelmuth,
Problem fixed, regards.

**Review by TylerHelmuth — APPROVED** [review]

**Pull request merged**